src/org/omegat/gui/scripting/ScriptingWindow.java

```diff
@@ -586,8 +597,56 @@ public void executeScriptFile(ScriptItem scriptItem, Map<String, Object> additio
             logResult(StringUtil.format(OStrings.getString("SCW_SCRIPT_LOAD_ERROR"), scriptItem.getFile()), e);
         }
     }
+    
 
-    public void executeScript(String scriptString, ScriptItem scriptItem, Map<String, Object> additionalBindings) {
+    /** Execute scripts sequentially to make sure they don't interrupt each other. */
+    public void executeScriptFiles(final ArrayList<ScriptItem> scriptItems, final Map<String, Object> bindings) {
+        for (ScriptItem scriptItem : scriptItems) {
+            try {
+                String scriptString = scriptItem.getText();
+                queuedWorkers.add(createScriptWorker(scriptString, scriptItem, bindings));
+            } catch (IOException e) {
+                // TODO: Do we really want to handle the exception here, like this?
+                // This method can be called in instances when the Scripting Window
+                // is not visible, so it might make more sense to let the caller
+                // handle the exception.
+                logResult(StringUtil.format(OStrings.getString("SCW_SCRIPT_LOAD_ERROR"), scriptItem.getFile()), e);
+            };
+        }
+
+        executeScriptWorkers();
+    }
```

Comment:
This looks like basically the same thing as `executeScriptFile(ScriptItem, Map)`; why not have that wrap its argument in a list and delegate to this `executeScriptFiles`?

Reply:
Also `ArrayList<ScriptItem>` is needlessly specific; it can be `List<ScriptItem>`. Then for my previous suggestion you can wrap with `Arrays.asList`.